registry/garbagecollect.go

```diff
@@ -0,0 +1,133 @@
+package registry
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/digest"
+	"github.com/docker/distribution/reference"
+	"github.com/docker/distribution/registry/storage"
+	"github.com/docker/distribution/registry/storage/driver"
+	"github.com/docker/distribution/registry/storage/driver/factory"
+
+	"github.com/spf13/cobra"
+)
+
+func markAndSweep(driver driver.StorageDriver) error {
+	ctx := context.Background()
+
+	// Construct a registry
+	registry, err := storage.NewRegistry(ctx, driver)
+	if err != nil {
+		return fmt.Errorf("Failed to construct registry: %v", err)
+	}
+
+	repositoryEnumerator, ok := registry.(distribution.RepositoryEnumerator)
+	if !ok {
+		return fmt.Errorf("Coercion error: %v", err)
```

Comment:
err will be nil here.  An error message explaining why this error occurred is probably better.